internal/event/context.go

```diff
@@ -27,18 +24,21 @@ type kubeContextsResponse struct {
 
 type ContextGeneratorOption func(generator *ContextsGenerator)
 
+type KubeContextStore interface {
+	CurrentContext() string
+	Contexts() []kubeconfig.Context
+}
+
 // ContextsGenerator generates kube contexts for the front end.
 type ContextsGenerator struct {
-	ConfigLoader kubeconfig.Loader
-	DashConfig   config.Dash
+	KubeContextStore KubeContextStore
 }
 
 var _ octant.Generator = (*ContextsGenerator)(nil)
 
-func NewContextsGenerator(dashConfig config.Dash, options ...ContextGeneratorOption) *ContextsGenerator {
+func NewContextsGenerator(kubeContextStore KubeContextStore, options ...ContextGeneratorOption) *ContextsGenerator {
 	kcg := &ContextsGenerator{
-		ConfigLoader: kubeconfig.NewFSLoader(),
-		DashConfig:   dashConfig,
+		KubeContextStore: kubeContextStore,
```

Comment:
Make sure the context is updating as the kubeconfig changes from the watcher. Current behavior has the context updating in Octant via `kubectl config use-context`. A similar case also holds if additional contexts are added though tools like kind.

Additionally, an instance of Octant running via `FromClientConfig` should not be looking at the filesystem.